torch/_subclasses/fake_tensor.py

```diff
@@ -1927,7 +1933,8 @@ def map_out(e):
         else:
             return e
 
-    return tree_map(map_out, r)
+    flat_out = [map_out(o) for o in flat_out]
+    return pytree.tree_unflatten(flat_out, out_spec)
```

Comment:
Here perhaps we just want to `tree_map` all the transformation on `r` rather than unpacking and packing again?